docroot/modules/custom/mass_alerts/src/Controller/AlertsController.php

```diff
@@ -0,0 +1,253 @@
+      '#cache' => [
+        'max-age' => 60,
+        'tags' => [
+          'node_list:alert'
+        ]
+      ],
+    ];
+
+    $output = $this->renderer->renderRoot($build);
+    $response = new CacheableResponse($output);
+    $response->addCacheableDependency(CacheableMetadata::createFromRenderArray($build));
+    return $response;
+  }
+
+  /**
+   * Returns the specific page alerts rendered.
+   */
+  public function handlePageRequest($nid, Request $request) {
+
+    $results = [
+      'headerAlerts' => [],
+    ];
+
+    $nodeStorage = $this->entityTypeManager()->getStorage('node');
+
+    $currentPage = $nodeStorage->load($nid);
+
+    if ($currentPage) {
+      $organizations = $currentPage->get('field_organizations')->getValue();
+      $org_ids = [];
+
+      foreach ($organizations as $org) {
+        $org_ids[] = $org['target_id'];
+      }
+
+      $query = $nodeStorage->getQuery();
+      $query->condition('type', 'alert');
+      $query->condition('status', 1);
+
+      $orContition = $query->orConditionGroup();
+      $orContition->condition('field_target_page.target_id', $nid);
+
+      if (!empty($org_ids)) {
+        $orContition->condition('field_target_organization.target_id', $org_ids, 'IN');
+      }
+
+      $query->condition($orContition);
+
+      $nids = $query->execute();
+
+      $nodes = $nodeStorage->loadMultiple(array_values($nids));
+
+      $alerts = [];
+
+      foreach ($nodes as $node) {
+
+        $changed_date = $node->getChangedTime();
+        $id = $node->uuid() . '__' . $changed_date;
+
+        $items = $node->get('field_alert')->referencedEntities();
+        $item = reset($items);
+
+        $link_type = $item->get('field_emergency_alert_link_type')->getString();
+
+        if ($link_type == '1') {
+          $uri = $item->get('field_emergency_alert_link')->getString();
+          if ($uri) {
+            $url = Url::fromUri($uri)->toString(TRUE)->getGeneratedUrl();
+          }
+        }
+        else {
+          $url = $node->toUrl()->toString(TRUE)->getGeneratedUrl();
+          $url .= '#' . $item->id();
+        }
+
+        $prefix = NULL;
+        $severity = $node->get('field_alert_severity')->getString();
+
+        if ($severity == 'informational_notice') {
+          $prefix = 'Notice';
+        }
+
+        $alert = [
+          'id' => $id,
+          'text' => $item->get('field_emergency_alert_message')->getString(),
+          'href' => ($link_type == '2') ? '' : $url,
+          'info' => '',
+        ];
+
+        if ($alert) {
+          $alert['prefix'] = $prefix;
+        }
+
+        $alerts[$changed_date] = $alert;
+      }
+
+      krsort($alerts);
+      $results['headerAlerts'] = array_values($alerts);
+    }
+
+    $build = [
+      '#theme' => 'mass_alerts_page',
+      '#headerAlerts' => $results['headerAlerts'],
+      '#cache' => [
+        'max-age' => Cache::PERMANENT,
+        'tags' => [
+          'node:' . $nid,
```

Comment:
Not sure what $nid is. I think you can omit this line, and instead add: `$response->addCacheableDependency($node)`. Same for sitewide alerts.